src/pages/DashboardPage/index.tsx

```diff
@@ -13,33 +12,44 @@ import {
   sortColumns,
   getCardsAfterDragAndDrop,
   getColumnsAfterDragAndDrop,
+  getNewColumn,
+  getNewCard,
 } from '../../utils';
 import { BoardAside } from '../../components/BoardAside';
 import { useAuth } from '../../contexts/Auth';
 import { useInput } from '../../hooks/useInput';
 import { BgContext } from '../../contexts/BgContext';
+import {
+  ColumnType,
+  CardType,
+  boardTitleProps,
+  ArrVisibleProps,
+} from './index.props';
 
-// TODO: Объект задачи отдельная сущность, массив с досками о
-//  тдельная сущность. Когда открываем страницу Dashboard мы должны сделать фетч всех досок
-
-// TODO: Колонка это отдельный компонент который должен сделать
-//  фетч всех задач и отрендерить в себе только те которые привязаны к ней
-
-// TODO: Постоянный перерендер. Исправить срочно!!!
 const DashboardPage = () => {
-  const [columns, setColumns] = useState([]);
-  const [cards, setCards] = useState([]);
-  const [cardsVisible, setCardsVisible] = useState([]);
+  const [columns, setColumns] = useState<ColumnType[] | []>([]);
+  const [cards, setCards] = useState<CardType[] | []>([]);
+  const [cardsVisible, setCardsVisible] = useState<ArrVisibleProps[] | []>([]);
   const [changeCardPos, setChangeCardPos] = useState(false);
   const { user, client } = useAuth();
   const { boardId } = useParams();
-  const [boardTitle, setBoardTitle] = useState({});
-  const inputSearch = useInput();
-  const { value } = inputSearch;
+  const initial: boardTitleProps = {
+    title: '',
+    username: '',
+  };
+  const [boardTitle, setBoardTitle] = useState(initial);
+  const inputSearch = useInput('') as {
+    value: string;
+    onChange: (
+      ev: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement> | undefined
+    ) => void;
+    onClear: () => void;
+  };
+
   const boardID = Number(boardId);
   const { changeWrapperBg } = useContext(BgContext);
 
-  const getData = async (type, id) => {
+  const getData = async (type: string, id: number | null) => {
```

Comment:
may be wrap this in useEffect? otherwise on each render of component this async function is run